mods/legacy-shell-message-boxes.wh.cpp

```diff
@@ -0,0 +1,256 @@
+// ==WindhawkMod==
+// @id              legacy-shell-message-boxes
+// @name            Legacy Shell Message Boxes
+// @description     Makes ShellMessageBox use message boxes rather than task dialogs
+// @version         1.0.0
+// @author          aubymori
+// @github          https://github.com/aubymori
+// @include         *
+// @compilerOptions -lshlwapi -lversion
+// ==/WindhawkMod==
+
+// ==WindhawkModReadme==
+/*
+# Legacy Shell Message Boxes
+This mod makes message boxes opened by the `ShellMessageBox` API use regular message boxes like in
+Windows XP and before instead of task dialogs, like they do in Windows Vista and later.
+
+**Before**:
+
+![Before](https://raw.githubusercontent.com/aubymori/images/refs/heads/main/legacy-shell-message-boxes-before.png)
+
+**After**:
+
+![After](https://raw.githubusercontent.com/aubymori/images/refs/heads/main/legacy-shell-message-boxes-after.png)
+*/
+// ==/WindhawkModReadme==
+
+#include <shlwapi.h>
+#include <windhawk_utils.h>
+
+DWORD g_dwSMBThreadID = (DWORD)-1;
```

Comment:
Optional suggestion: Instead of using shared value which can be overridden by different threads, use something like:
```cpp
thread_local bool g_inSMB = false;
```